flang/lib/Lower/CallInterface.cpp

```diff
@@ -166,9 +171,27 @@ bool Fortran::lower::CalleeInterface::hasAlternateReturns() const {
 }
 
 std::string Fortran::lower::CalleeInterface::getMangledName() const {
-  return funit.isMainProgram()
-             ? fir::NameUniquer::doProgramEntry().str()
-             : converter.mangleName(funit.getSubprogramSymbol());
+
+  if (funit.isMainProgram())
+    return fir::NameUniquer::doProgramEntry().str();
+  // Do NOT mangle names for functions/subroutines with BIND(C) attribute.
+  // These could be potential calls to external RTL (i.e OpenMP), that may not
+  // follow the same mangling semantics.
+  if (Fortran::semantics::IsBindCProcedure(funit.getSubprogramSymbol())) {
+    const auto &symbol = funit.getSubprogramSymbol();
+    if (const auto *procDetails{
```

Comment:
Thanks @jeanPerier for clarification regarding the call. I've made relevant changes, however I'm not sure WRT to this change i.e `ProcEntity` . One more thing I noticed is 
for following snippet name mangling is still happening
```
function foo()
  external :: omp_get_num_threads
  foo1 = omp_get_num_threads()
end function
$ bbc -fopenmp -emit-fir test.f90 -o - | egrep "num_threads"
%2 = fir.call @_QPomp_get_num_threads() : () -> f32
```

Reply:
The mangling you are showing seems as expected to me. `omp_get_num_threads` is not a reserved name. You have to give it the BIND(C) attribute if you want it to not be mangled.
I am pretty sure every compiler would mangle the function with a trailing `_` for your example.

Here, the "correct" fortran would probably be:
```
function foo()
  interface
    integer function omp_get_num_threads() bind(C)
    end function
  end interface
  foo1 = omp_get_num_threads()
end function
```
I am putting "correct", because I think people should do "use omp_lib, only: omp_get_num_threads" if they just want omp_get_num_threads instead of redefining the interface manually.

Reply:
Got it! Thanks for correcting my understanding!
I've added 1 simple test too. Any more overall thoughts WRT this PR ?